kahuna/public/js/search/query.js

```diff
@@ -62,6 +69,16 @@ query.controller('SearchQueryCtrl',
     Object.keys($stateParams)
           .forEach(setAndWatchParam);
 
+    ctrl.changeQuery = () =>
+        searchQueryService.set(ctrl.filter.query);
+
+    // FIXME: This is here to stop circular injection of the model.
+    // Avoiding: queryChange => set() => emit() => queryChange()
+    searchQueryService.q$.filter(q =>
+        q !== ctrl.filter.query
```

Comment:
* User types `"culture"`
* `q$.emit("culture")`
* `filter.query = q`

We just set q twice unessecarily, which could also cause problems. This is the same [challenge we had in the usage rights editor](https://github.com/guardian/grid/blob/master/kahuna/public/js/usage-rights/usage-rights-editor.js#L47-L58)